File: Java/Arrays/Insercion_inicio.java
/*
Arrays en Java,Insercion en el primer elemento de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 04/09/2026
*/

public class Insercion_inicio {
    public static void main(String[] args){
        int[] input_arr = {11, 21, 31, 41, 51, 61};     //array estatico

        System.out.println("Antes de la insercion, el array es: ");

        for (int elemento : input_arr){
            System.out.println(elemento);
        }

        System.out.println("Despues de la insercion, el array es: ");

        int[] input_arr2 = new int[input_arr.length +1];        //crea un nuevo array con el tamaño del original +1 
        input_arr2[0] = 1;      //en el indice 0 se le da el valor de 1
        System.arraycopy(input_arr, 0, input_arr2, 1, input_arr.length);        //se copia el primer array al segundo, (array1_origen, indice donde empieza a copiar, array2_destino, indice desde donde se empieza a copiar, numero de elementos del array1 

        for (int elemento : input_arr2){
            System.out.println(elemento);
        }
        
    }
}

File: C++/Arrays/Insercion_inicio.cpp
/*
Arrays en C++,Insercion en el primer elemento de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 06/09/2026
*/

#include <iostream>
#include <vector>

using namespace std;
int main(){
    int input_arr [] = {5, 15, 25, 35, 45, 55};
    int tamano = sizeof(input_arr)/sizeof(input_arr[0]);
    int elem {77};

    cout << "Antes de la insercion, el array es: ";
    for (int elemento : input_arr){     //for each
        cout << elemento << " ";
    }

    //Nuevo arreglo
    int new_input_arr [tamano + 1];

    //Insercion
    for (int i = 1; i <= tamano; i++ ){
        new_input_arr[i] = input_arr[i-1];
    }
    new_input_arr[0] = elem;

    cout << "\nDespues de la insercion, el array es: ";
    for (int elemento : new_input_arr){
        cout << elemento << " ";
    }
    
    //otra manera seria
    vector <int> array = {5, 15, 25, 35, 45, 55};

    array.insert(array.begin() + 0, elem);          //nombredelarreglo.insert("es el inicio del arreglo" + desplazamiento, elemento a agregar)

    cout << "\n\nDespues de la insercion, el array es: ";
    for (int elemento : array){
        cout << elemento << " ";
    }

    return 0;
}